linode/rdns/framework_resource.go

```diff
@@ -0,0 +1,149 @@
+package rdns
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_rdns",
+				Schema: &frameworkResourceSchema,
+				IDType: types.StringType,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateOpts := linodego.IPAddressUpdateOptions{
+		RDNS: data.RDNS.ValueStringPointer(),
+	}
+
+	ip, err := client.UpdateIPAddress(ctx, data.Address.ValueString(), updateOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create Linode RDNS",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseIP(ip)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	client := r.Meta.Client
+
+	var data ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ip, err := client.GetIPAddress(ctx, data.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to read the Linode RDNS", err.Error(),
+		)
+		return
+	}
+
+	data.parseIP(ip)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var state, plan ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	var updateOpts linodego.IPAddressUpdateOptions
+
+	resourceUpdated := false
+
+	if !state.RDNS.Equal(plan.RDNS) {
+		updateOpts.RDNS = plan.RDNS.ValueStringPointer()
+		resourceUpdated = true
+	}
+
+	if resourceUpdated {
+		ip, err := client.UpdateIPAddress(ctx, state.Address.ValueString(), updateOpts)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Failed to update the Linode RDNS",
+				err.Error(),
+			)
+			return
+		}
+
+		state.parseIP(ip)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
```

Comment:
> An error is returned unless every null or known value in the request plan is saved exactly as-is into the response state. Only unknown plan values can be modified.

https://developer.hashicorp.com/terraform/plugin/framework/resources/update#caveats
I think we should save `plan`, not API response `ip` into the state here.